lib/rdfc-processor.ts

```diff
@@ -69,10 +71,16 @@ export async function processor(
         streamId ? df.namedNode(streamId) : undefined,
     );
 
-    client.on("fragment", (fragment) => logger.verbose(`Got fragment: ${fragment.id.value}`));
-
     const reader = client.stream({ highWaterMark: 10 }).getReader();
 
+    client.on("fragment", async (fragment) => {
+        logger.verbose(`Got fragment: ${fragment.url}`);
+        if (interruptOnFragment?.includes(fragment.url)) {
+            logger.info(`Interrupting on fragment ${fragment.url}`);
+            await reader.cancel();
+        }
+    });
```

Comment:
Do we want to keep this in here?

Reply:
I agree it is not an elegant solution, mainly because this was useful only for the tests, but I couldn't think of a better alternative that allowed me to interrupt the stream on specific fragments. The rdf-connect processor function does not return the `ReadableStream` where this can be called, and it only returns the `Client`, where we could implement something that allows to interrupt, only after the processed has finished.

A pragmatic solution could be not to test this via the rdf-connect processor but directly on the client. Any other ideas @smessie?